com.civicrm.hrjobroles/tests/phpunit/HrJobRolesTestBase.php

```diff
@@ -98,6 +108,27 @@ public function createSampleOptionGroupsAndValues()  {
       $params = ['option_group_id' => $groupID, 'name' => $value, 'value' => $value ];
       CRM_Core_BAO_OptionValue::create($params);
     }
+
+  }
+
+  /**
+   * Find and retrieve job role by any of its properties
+   *
+   * @param int|string $value
+   * @param string $identifier
+   *
+   * @return \CRM_Hrjobroles_BAO_HrJobRoles|NULL
+   */
+  public function findRoleByProperty($value, $identifier = 'id')  {
```

Comment:
> but in every call you're always using the ID

no I have used the title in some places. but I wasn't aware of commonRetrieve method ... mmmm honestly I want to avoid as much as possible to re run the tests again (they take a lot of time) , so if I didn't made another changes that may require me to run the tests again I will keep it like that for now.